pkg/costmodel/costmodel.go

```diff
@@ -501,14 +501,14 @@ func (cm *CostModel) ComputeCostData(cli prometheusClient.Client, cp costAnalyze
 
 				gpuReqCount := 0.0
 				if g, ok := container.Resources.Requests["nvidia.com/gpu"]; ok {
-					gpuReqCount = float64(g.Value())
+					gpuReqCount = g.AsApproximateFloat64()
```

Comment:
It is worth noting that [this can lose precision](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Quantity.AsApproximateFloat64). I suspect that this is fine for GPUs as extremely small fractional GPU (0.000005% of a GPU?) allocations is an unlikely use case; thus, the precision we will lose is probably negligible.